<a href="https://colab.research.google.com/github/dedemasutti/AI-projects/blob/master/Recognition_of_car_parts_model.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Código de análise e reconhecimento de peças automotivas com Keras e OpenCV

In [2]:
pip install opencv-python

In [3]:
import cv2
import numpy as np
from keras.applications.inception_v3 import InceptionV3, preprocess_input, decode_predictions

In [ ]:
# Carrega o modelo pré-treinado InceptionV3
model = InceptionV3(weights='imagenet')


In [ ]:
# Função para fazer a análise e reconhecimento de uma imagem
def analisar_imagem(imagem):
    # Carrega a imagem e redimensionar para o tamanho esperado pelo modelo
    img = cv2.imread(imagem)
    img = cv2.resize(img, (299, 299))

    # Pré-processamento da imagem para ser compatível com o modelo
    img = preprocess_input(img)

    # Adiciona uma dimensão extra para representar o batch
    img = np.expand_dims(img, axis=0)

    # Realiza a predição da imagem usando o modelo
    preds = model.predict(img)

    # Decodifica as predições em formato legível
    decoded_preds = decode_predictions(preds, top=3)[0]

In [4]:


    # Imprime as predições
    for (i, (class_id, class_name, prob)) in enumerate(decoded_preds):
        print(f"{i+1}. {class_name}: {prob*100:.2f}%")

# Exemplo de uso da função com uma imagem
analisar_imagem("/content/roda mercedes.jpg")


35363/35363 [==============================] - 0s 0us/step
1. car_wheel: 60.20%
2. disk_brake: 32.34%
3. unicycle: 0.42%


#####A imagem foi de uma roda de carro, e a resposta do modelo preditivo foi:
1. car_wheel: 60.20%
2. disk_brake: 32.34%
3. unicycle: 0.42%

Ou seja, 60,2% de que seja a roda de um carro.